pkg/machine/hyperv/stubber.go

```diff
@@ -55,12 +55,20 @@ func (h HyperVStubber) CreateVM(opts define.CreateVMOpts, mc *vmconfigs.MachineC
 		Memory:   uint64(mc.Resources.Memory),
 	}
 
-	networkHVSock, err := vsock.NewHVSockRegistryEntry(mc.Name, vsock.Network)
-	if err != nil {
-		return err
-	}
+	if opts.UserModeNetworking {
+		networkHVSock, err := vsock.NewHVSockRegistryEntry(mc.Name, vsock.Network)
+		if err != nil {
+			return err
+		}
 
-	mc.HyperVHypervisor.NetworkVSock = *networkHVSock
+		mc.HyperVHypervisor.NetworkVSock = *networkHVSock
+		mc.HyperVHypervisor.UserModeNetworking = true
+		h.UserModeNetworking = true
```

Comment:
This is slightly confusing, ` mc.HyperVHypervisor` and `HyperVStubber` both have the `vmconfigs.HyperVConfig` type, but I think they are 2 different instances.
Before this PR, nothing in the code changes or read the content of `h`
This needs to be stored in `h` as it’s needed by `UseProviderNetworkSetup` implementation.

Wondering if this `UserModeNetworking` should be only added to the `HypervStubber` struct insteadk but that’s most likely a matter of preference (ie I leave that up to you)

Reply:
right, updated the PR to only add `UserModeNetworking` to `HypervStubber` struct, nothing was reading the value of `HypervConfig.UserModeNetworking`